spec/lib/query/query_supporters_spec.rb

```diff
@@ -458,6 +458,155 @@
     end
   end
 
+  describe '.dupes_on_last_name_and_address_and_email' do
+    subject { QuerySupporters.dupes_on_last_name_and_address_and_email(np.id) }
+
+    it 'finds supporters with the same last name and address and email' do
+      supporter_1 = force_create(:supporter, nonprofit_id: np.id, name: 'Cacau Borges', address: 'Clear Waters Avenue', email: "email@example.com")
+      supporter_2 = force_create(:supporter, nonprofit_id: np.id, name: 'Penelope Borges', address: 'Clear Waters Avenue', email: "email@example.com")
+
+      expect(subject).to eq([[supporter_1.id, supporter_2.id]])
+    end
+
+    context 'when different names' do
+      it 'does not find' do
+        supporter_1 = force_create(:supporter, nonprofit_id: np.id, name: 'Cacau Borges', address: 'Clear Waters Avenue', email: "email@example.com")
+        supporter_2 = force_create(:supporter, nonprofit_id: np.id, name: 'Penelope Schultz', address: 'Clear Waters Avenue', email: "email@example.com")
+
+        expect(subject).to match_array([])
+      end
+    end
+
+    context 'when different addresses' do
```

Comment:
I don't know that we gain much by pulling the supporters up since there's only one test.

At the same time, I think it's helpful to keep the `context` and `it` separate in this case. If someone is skimming the context names, you get a rough sense of the different situations to consider when trying to understand `QuerySupporters.dupes_on_last_name_and_address_and_email`. If I combine and rewrite, I go from:

```ruby
context 'when different addresses' do
    it 'does not find' do
```
 
to:

```ruby
it 'does not find when different addresses' do
```

I think that's harder to parse all of them quickly because the meaningful difference between those different situations is "when different addresses".

Given all that, I'm inclined to keep as-is.